chat-widget/src/hooks/useCheckConversationValidity.ts

```diff
@@ -0,0 +1,58 @@
+import { LogLevel, TelemetryEvent } from "../common/telemetry/TelemetryConstants";
+
+import { Dispatch } from "react";
+import { ILiveChatWidgetAction } from "../contexts/common/ILiveChatWidgetAction";
+import { ILiveChatWidgetContext } from "../contexts/common/ILiveChatWidgetContext";
+import { LiveChatWidgetActionType } from "../contexts/common/LiveChatWidgetActionType";
+import { LiveWorkItemState } from "../common/Constants";
+import { TelemetryHelper } from "../common/telemetry/TelemetryHelper";
+import { getConversationDetailsCall } from "../common/utils";
+import useChatContextStore from "./useChatContextStore";
+import useChatSDKStore from "./useChatSDKStore";
+
+const useCheckConversationValidity = () => {
+    const [state, dispatch]: [ILiveChatWidgetContext, Dispatch<ILiveChatWidgetAction>] = useChatContextStore();
+    // eslint-disable-next-line @typescript-eslint/no-explicit-any
+    const chatSDK: any = useChatSDKStore();
+
+    const checkConversationValidity = async () => {
+        const requestIdFromCache = state.domainStates?.liveChatContext?.requestId;
+        // eslint-disable-next-line @typescript-eslint/no-explicit-any
+        let conversationDetails: any = undefined;
+
+        //Preserve current requestId
+        const currentRequestId = chatSDK.requestId ?? "";
+        dispatch({ type: LiveChatWidgetActionType.SET_INITIAL_CHAT_SDK_REQUEST_ID, payload: currentRequestId });
+
+        try {
+            chatSDK.requestId = requestIdFromCache;
+            conversationDetails = await getConversationDetailsCall(chatSDK);
```

Comment:
if requestIdFromCache is null or empty, do we still need to call GetConversationDetail ?